### DISCUSSION TITLE
Accessing  page permalinks in themes and plugins

### DISCUSSION THREAD

**Discussion by Mister-Hope** [vuepress/core#44] [Q&A] [answered]
Plugins like `seo`, `sitemap`  or `feed` need to access the page permalink.

But from the source code, I need to provide the PageOption

https://github.com/vuepress/vuepress-next/blob/55f780c4a516b2c0119328874906647af80ac73b/packages/%40vuepress/core/src/page/resolvePagePermalink.ts#L8-L16

From source its shape is like:

https://github.com/vuepress/vuepress-next/blob/55f780c4a516b2c0119328874906647af80ac73b/packages/%40vuepress/core/src/types/page.ts#L108-L118

But it's only pass in these two places:

https://github.com/vuepress/vuepress-next/blob/55f780c4a516b2c0119328874906647af80ac73b/packages/%40vuepress/core/src/app/createAppPages.ts#L16-L22

https://github.com/vuepress/vuepress-next/blob/55f780c4a516b2c0119328874906647af80ac73b/packages/%40vuepress/core/src/app/createAppPages.ts#L26-L34

It means currently V2 can not support global permallink like [V1](https://v1.vuepress.vuejs.org/zh/guide/permalinks.html#%E9%85%8D%E7%BD%AE%E6%B0%B8%E4%B9%85%E9%93%BE%E6%8E%A5)

And appOption is not containing it:

https://github.com/vuepress/vuepress-next/blob/55f780c4a516b2c0119328874906647af80ac73b/packages/%40vuepress/core/src/types/app/options.ts#L9-L46

I saw you are focusing on docs and bug fixes now, so is it a unfinished feature, or V2 wants to drop it?

**Comment by meteorlxy** [comment, marked as answer]
The "global permalink " should be a missing feature.

It's not documented in the config reference of v1: https://vuepress.vuejs.org/config

You know the docs and code of v1 is ... not good enough 😓 . So we have to figure out features from it, and it's likely to miss something.

**Reply by Mister-Hope** [reply]
https://v1.vuepress.vuejs.org/zh/guide/permalinks.html#%E9%85%8D%E7%BD%AE%E6%B0%B8%E4%B9%85%E9%93%BE%E6%8E%A5

**Reply by Mister-Hope** [reply]
So you will add it right? I saw you write tests about permalinks, and global options is passed (actual code is not though)

**Reply by meteorlxy** [reply]
I just noticed that guide. But the reference **should** be the full list of features / options, or you have to search them in every pages

**Reply by Mister-Hope** [reply]
Just as a reminder in case you forget it, vuepress do have a lot of features and options.😊

**Reply by meteorlxy** [reply]
> So you will add it right? I saw you write tests about permalinks, and global options is passed (actual code is not though)

I'll check it again. It's not complicated and we only need to add a global option and pass forward it when creating pages

**Reply by meteorlxy** [reply]
Since v2.0.0-alpha.24, we can use `extendsPageOptions` hook for that

**Comment by Mister-Hope** [comment]
@meteorlxy I am still unable to get the permallink.

> Plugins like seo, sitemap or feed need to access the page permalink.

I am not wanting the pattern, but the final permallink in plugins list above.

**Reply by meteorlxy** [reply]
What do you mean by "the final permallink"? The final page route path does not meet your needs?

https://github.com/vuepress/vuepress-next/blob/5d3925b0137177a6440bb8150a8f1285dceaa72f/packages/%40vuepress/core/src/page/resolvePagePath.ts#L6-L14

**Reply by Mister-Hope** [reply]
Wait, so if a page frontmatter 's permalink option is set, the filePath infered route is lost?🧐 I am sure that's not people are expecting. How can they index the pages with permalink (such as by time)

I am not sure, but in my memory, both of them is available in V1.

**Reply by meteorlxy** [reply]
`page.pathInferred`

Check the types of page: https://github.com/vuepress/vuepress-next/blob/5d3925b0137177a6440bb8150a8f1285dceaa72f/packages/%40vuepress/core/src/types/page.ts#L4-L7
